src/Server/ProcessMessage.idr

```diff
@@ -228,8 +228,8 @@ loadURI conf uri version = do
   update LSPConf ({ virtualDocuments $= insert uri (fromMaybe 0 version, res ++ "\n") })
   --     A hack to solve some interesting edge-cases around missing newlines ^^^^^^^
   setSource res
-  errs <- catch
-            (buildDeps fname)
+  errs <- catch (do mods <- getBuildMods EmptyFC [] fname
+                    buildMods EmptyFC 1 (length mods) mods)
```

Comment:
Best I can tell, this change breaks the LSP server (as a Neovim user, at least). No useful errors coming out of it from a cursor look over the LSP logs, just a lack of the LSP reporting anything actionable to the editor so you never get highlighting (or, naturally, any other LSP features).

I double checked by undoing only this line's change and saw the server work (again, this is in Neovim).

Reply:
That's odd. Maybe vim is interacting with the LSP differently than vscode.  vscode is working for me. 

I'm not able to reproduce this locally in nvim, but I realized I've got edwin's `idris2-vim`, which I don't think is LSP based.  I'll see if I can get nvim set up with LSP and sort out the issue.  Marking this PR as draft for now.  Thanks for taking a look.

(I'm also not using LSP with emacs, but I don't know if anybody does for Idris.)

Reply:
ok, I see something now in vscode. The testing that I did last night was with a file that had an false coverage error in addition to warnings. I did see the warnings, the warnings went away if I fixed them, and the editor functions work, but not if I try it with an error-free file, the LSP functionality is not there. Maybe the block that reloads the file in case of error is fixing something.  I'll investigate.